nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/items/flow/local-changes-dialog/local-changes-table/local-changes-table.ts

```diff
@@ -0,0 +1,174 @@
+    componentId: string;
+    componentName: string;
+    processGroupId: string;
+    differenceType: string;
+    difference: string;
+}
+
+@Component({
+    selector: 'local-changes-table',
+    standalone: true,
+    imports: [MatFormField, MatInput, MatLabel, ReactiveFormsModule, MatTableModule, MatSortModule],
+    templateUrl: './local-changes-table.html',
+    styleUrl: './local-changes-table.scss'
+})
+export class LocalChangesTable implements AfterViewInit {
+    private destroyRef: DestroyRef = inject(DestroyRef);
+    initialSortColumn: 'componentName' | 'changeType' | 'difference' = 'componentName';
+    initialSortDirection: 'asc' | 'desc' = 'asc';
+
+    filterTerm = '';
+    totalCount = 0;
+    filteredCount = 0;
+
+    activeSort: Sort = {
+        active: this.initialSortColumn,
+        direction: this.initialSortDirection
+    };
+
+    displayedColumns: string[] = ['componentName', 'changeType', 'difference', 'actions'];
+    dataSource: MatTableDataSource<LocalChange> = new MatTableDataSource<LocalChange>();
+    filterForm: FormGroup;
+
+    @Input() set differences(differences: ComponentDifference[]) {
+        const localChanges: LocalChange[] = this.explodeDifferences(differences);
+        this.dataSource.data = this.sortEntities(localChanges, this.activeSort);
+        this.dataSource.filterPredicate = (data: LocalChange, filter: string) => {
+            const { filterTerm } = JSON.parse(filter);
+            // check the filter term in both the name and type columns
+            return (
+                this.nifiCommon.stringContains(data.componentName, filterTerm, true) ||
+                this.nifiCommon.stringContains(data.differenceType, filterTerm, true)
+            );
+        };
+        this.totalCount = localChanges.length;
+        this.filteredCount = localChanges.length;
+
+        // apply any filtering to the new data
+        const filterTerm = this.filterForm.get('filterTerm')?.value;
+        if (filterTerm?.length > 0) {
+            this.applyFilter(filterTerm);
+        }
+    }
+
+    @Output() goToChange: EventEmitter<NavigateToComponentRequest> = new EventEmitter<NavigateToComponentRequest>();
+
+    constructor(
+        private formBuilder: FormBuilder,
+        private nifiCommon: NiFiCommon
+    ) {
+        this.filterForm = this.formBuilder.group({ filterTerm: '', filterColumn: 'componentName' });
+    }
+
+    ngAfterViewInit(): void {
+        this.filterForm
+            .get('filterTerm')
+            ?.valueChanges.pipe(debounceTime(500), takeUntilDestroyed(this.destroyRef))
+            .subscribe((filterTerm: string) => {
+                this.applyFilter(filterTerm);
+            });
+    }
+
+    applyFilter(filterTerm: string) {
+        this.dataSource.filter = JSON.stringify({ filterTerm });
+        this.filteredCount = this.dataSource.filteredData.length;
+    }
+
+    formatComponentName(item: LocalChange): string {
+        return item.componentName;
```

Comment:
For a funnel this displays 'undefined':
![Screenshot 2024-04-09 at 5 38 08 PM](https://github.com/apache/nifi/assets/6797571/b2dea080-30fc-4f6d-9c95-5b6eb744dde9)